### PR TITLE
fix: copy tooltips

### PR DESCRIPTION
**Pull request by CzarekDryl**
## Description

- Added tooltip dedicated for copying
- Copy tooltip should hide after 2 seconds or on mouse leave/clicked outside
- Changed old component to new one in all places that it was used

Resolves [https://github.com/JoinColony/colonyCDapp/issues/2142](https://github.com/JoinColony/colonyCDapp/issues/2142)

### CODE REVIEW

File: src/components/v5/shared/CopyTooltip/CopyTooltip.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        triggerRef && !triggerRef.contains(event.target as Node);
+
+      if (isOutsideTrigger) {
+        setIsTooltipVisible(false);
+      }
+    };
+
+    if (visible) {
+      document.addEventListener('click', handleClickOutside);
+      document.addEventListener('mousemove', handleMouseLeave);
```
Comment (jakubcolony) [inline src/components/v5/shared/CopyTooltip/CopyTooltip.tsx:79]: please rename the handler to match the event

File: src/components/v5/shared/CopyTooltip/types.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  placement?: PlacementType;
+  offset?: [number, number];
+  popperOptions?: PopperOptions;
+  interactive?: boolean;
+  showArrow?: boolean;
+  isOpen?: boolean;
+  isSuccess?: boolean;
+  isFullWidthContent?: boolean;
+  className?: string;
+  selectTriggerRef?: (ref: HTMLElement | null) => HTMLElement | null;
```
Comment (jakubcolony) [inline src/components/v5/shared/CopyTooltip/types.ts:16]: This could extend the generic tooltip props

**Review by jakubcolony — CHANGES_REQUESTED** [review]
The colony address copy tooltip works as intended and it closes on mouse leave/click:
Screenshot 2024-04-10 at 15 52 51
03bf41

But it looks like the one on extensions page got broken:

https://github.com/JoinColony/colonyCDapp/assets/112586815/d81fb779-5f5a-46bf-95de-533c031d18f4

File: src/components/v5/shared/CopyTooltip/CopyTooltip.tsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+      document.removeEventListener('click', handleClickOutside);
+      document.removeEventListener('mousemove', handleMouseLeave);
+    };
+  }, [tooltipRef, triggerRef, visible, setIsTooltipVisible]);
+
+  return (
+    <>
+      <div
+        className={clsx(className, 'flex', {
+          'w-full': isFullWidthContent,
```
Comment (bassgeta) [inline src/components/v5/shared/CopyTooltip/CopyTooltip.tsx:95]: i would maybe just ditch this prop since we can pass w-full in the className if need be

**Review by bassgeta — CHANGES_REQUESTED** [review]
On top of what Jakub already mentioned, would it be an option that this component wraps around or existing `Tooltip`? It's the same component with custom colors and a bit more elaborate logic for closing it.

**Review by melyndav — DISMISSED** [review]
Hey @CzarekDryl thanks for working on improving this component.

Everything looks great. The only issue I have is that every tooltip has an ! added to the copy which is not wanted or in Figma. But I'll log this for another day.

**Review by ArmandoGraterol — DISMISSED** [review]
LGTM 👍 From what I can see even the one in the extension page is working like it should now, nice job!
Screenshot 2024-04-25 at 17 22 45

**Review by jakubcolony — DISMISSED** [review]
LGTM, thanks for the changes @CzarekDryl 👍

**Comment by CzarekDryl** [comment]
@bassgeta Please take a look at this one

**Review by arrenv — CHANGES_REQUESTED** [review]
@CzarekDryl Thank you for updating this. I should have been more specific in the spec, the mouseleave and onclick close should be on the tooltip, not on the icon.

**Review by jakubcolony — APPROVED** [review]

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
I'm now experiencing this bug with the tooltip where the usual hover tooltip stays visible after clicking and moving the cursor away.


https://github.com/JoinColony/colonyCDapp/assets/34915414/eb459294-9dc5-4c37-8352-4b8919f7f207

**Comment by arrenv** [comment]
@CzarekDryl Are we able to close this out?

**Pull request closed without merging**